src/jquery.interactive-world-map.js

```diff
@@ -1,142 +1,180 @@
-$(function(){
-
-    $.fn.interactiveWorldMap = function(options) {
-        event.preventDefault();
-        var marginCountryPopupForTop  = $.fn.interactiveWorldMap.defaults.marginCountryPopupForTop;
-        var marginCountryPopupForLeft = $.fn.interactiveWorldMap.defaults.marginCountryPopupForLeft;
-        var dataCountries = {};
-        if(options) {
-            dataCountries                           = options['countries'];
-            //Location of the popup when its launched
-            var margins                             = options['margins'];
-
-            if(margins && typeof(margins['marginCountryPopupForTop']) != "undefined" ) {
-                marginCountryPopupForTop            = margins['marginCountryPopupForTop'];
-            }
+;$(function( $, window, document, undefined) {
+
+    var pluginName = "interactiveWorldMap";
+    var generalColor = '#cccccc';
+    var marginCountryPopupForLeft;          //distance between the popup launched when you click a country and the left of the browser
+    var marginCountryPopupForTop;           //distance between the popup launched when you click a country and the top  of the browser
+    var dataCountries = {};                 //Hold the countries information sent by parameters
+    var countriesFill = {};                 //Hold the countries code that must be Highlighted for be a YFU Country
+    var highlightColor;                     //The #color used for highlighted  the countries
+    var hoverPattern;                       //The svg pattern (html code) that uses the hover colors over the YFU countries and make it dotted.
+    var hoverColor;                         //The hover color for the YFU countries. It's used by patternHover.
+    var yfuCountriesHighlighted;            //The pattern that makes the yfu countries dotted and highlighted
+    var generalDottedPatternStyle;          //The general dotted style for all the countries
+    /*  Holds the patterns sizes of the map */
+    var _cx;
+    var _cy;
+    var _r;
+    var _x;
+    var _y;
+
+    var defaults = {
+        hovercoloroption : "#FF0000",
+        highlightedcoloroption: "#848484",
+        marginCountryPopupForTop: 100,
+        marginCountryPopupForLeft: 60
+    };
```

Comment:
The whole variable section is still way to convoluted. Use hashes to capsule things that belong together and have one set of default values for all things that you need. If you structure it in a good hash, you can properly work with `$.extend` and don't need to manually assign stuff, for example:

```
var defaults = {
  popup: {
    margins: { top: 100, left: 60 },
    borderColor: 'green'
  }
  colors: {
    hover: 'red',
    highlighted: 'blue'
  }
  //...
}

// fetch options
// fetch data_options
// and finally merge verything in that order
var settings = $.extend(defaults, options, data_options)

// after this you can access all your information through `this.settings` in your entire plugin, instead of having to use 100 different variables.